### ISSUE TITLE
Function clause error in PositionHelper when run with Elixir 1.11.0-rc.0

### ISSUE THREAD

**Issue by axelson** [rrrene/credo#805]
### Environment

* Credo version (`mix credo -v`): 1.4.0
* Erlang/Elixir version (`elixir -v`): Elixir 1.11.0-rc.0 (48cee5a) (compiled with Erlang/OTP 23)
* Operating system:

### What were you trying to do?

Run credo on my code

### Expected outcome

Run without an exception

### Actual outcome

```
Checking 323 source files (this might take a while) ...
** (config) Error while running Elixir.Credo.Check.Consistency.SpaceAroundOperators

15:56:41.084 [error] Task #PID<0.608.0> started from #PID<0.94.0> terminating
** (FunctionClauseError) no function clause matching in Credo.Code.Token.PositionHelper.position/1
    lib/credo/code/token.ex:49: Credo.Code.Token.PositionHelper.position({:sigil, {45, 11, nil}, 85, ["2020-09-18 12:00:00Z"], [], nil, "["})
    lib/credo/check/consistency/space_around_operators/space_helper.ex:58: Credo.Check.Consistency.SpaceAroundOperators.SpaceHelper.no_space_between?/2
    lib/credo/check/consistency/space_around_operators/collector.ex:166: Credo.Check.Consistency.SpaceAroundOperators.Collector.without_space?/3
    lib/credo/check/consistency/space_around_operators/collector.ex:130: Credo.Check.Consistency.SpaceAroundOperators.Collector.record_spaces/4
    lib/credo/check/consistency/space_around_operators/collector.ex:37: Credo.Check.Consistency.SpaceAroundOperators.Collector.traverse_tokens/3
    lib/credo/check/consistency/collector.ex:140: anonymous fn/3 in Credo.Check.Consistency.Collector.find_issues/4
    (elixir 1.11.0-rc.0) lib/enum.ex:1399: Enum."-map/2-lists^map/1-0-"/2
    (elixir 1.11.0-rc.0) lib/enum.ex:1399: Enum."-map/2-lists^map/1-0-"/2
Function: #Function<10.46020751/0 in Credo.Check.Runner.run/2>
    Args: []
** (EXIT from #PID<0.94.0>) an exception was raised:
    ** (FunctionClauseError) no function clause matching in Credo.Code.Token.PositionHelper.position/1
        lib/credo/code/token.ex:49: Credo.Code.Token.PositionHelper.position({:sigil, {45, 11, nil}, 85, ["2020-09-18 12:00:00Z"], [], nil, "["})
        lib/credo/check/consistency/space_around_operators/space_helper.ex:58: Credo.Check.Consistency.SpaceAroundOperators.SpaceHelper.no_space_between?/2
        lib/credo/check/consistency/space_around_operators/collector.ex:166: Credo.Check.Consistency.SpaceAroundOperators.Collector.without_space?/3
        lib/credo/check/consistency/space_around_operators/collector.ex:130: Credo.Check.Consistency.SpaceAroundOperators.Collector.record_spaces/4
        lib/credo/check/consistency/space_around_operators/collector.ex:37: Credo.Check.Consistency.SpaceAroundOperators.Collector.traverse_tokens/3
        lib/credo/check/consistency/collector.ex:140: anonymous fn/3 in Credo.Check.Consistency.Collector.find_issues/4
        (elixir 1.11.0-rc.0) lib/enum.ex:1399: Enum."-map/2-lists^map/1-0-"/2
        (elixir 1.11.0-rc.0) lib/enum.ex:1399: Enum."-map/2-lists^map/1-0-"/2
```

Unfortunately this code isn't open source so I can't share it. I haven't yet tested on another project. Also I wanted to mention that I realize that 1.11-rc.0 probably isn't expected to be supported by Credo just yet.

**Comment by eksperimental** [comment]
Same here

**Comment by rrrene** [comment]
This should be fixed in `1.5.0-rc.2` - could you give it a try?

**Comment by axelson** [comment]
@rrrene Confirmed that it works on my codebase 👍 

And the new warning `Module attribute @some_attribute makes use of unsafe Application configuration call Application.fetch_env!/2` is a nice touch. I should probably go ahead and migrate to `compile_env` since I'm on 1.10 already.

**Comment by eksperimental** [comment]
I think this issue can be closed

**Comment by skovmand** [comment]
I also get this error on elixir 1.11.0-otp-23. Was it fixed by `1.5.0`?

**Comment by eksperimental** [comment]
> I also get this error on elixir 1.11.0-otp-23. Will it be fixed in `1.5.0`?

did you try Credo v1.5.0-rc.2 ?

**Comment by skovmand** [comment]
I tried to use it, but I got confused by Credo requiring the package `file_system` [link](https://hex.pm/packages/file_system) (a file system watcher), so I backed out. What does credo need that for? (Anyway, I know that discussion is not related to this issue)

**Comment by eksperimental** [comment]
> I tried to use it, but I got confused by Credo requiring the package `file_system` [link](https://hex.pm/packages/file_system) (a file system watcher), so I backed out. What does credo need that for? (Anyway, I know that discussion is not related to this issue)

```console
$ git blame mix.exs | grep file_system
a95a404cb (René Föhring   2020-10-04 09:51:29 +0200 117)       extra_applications: [:bunt, :crypto, :eex, :ex_unit, :file_system, :inets, :jason, :logger]
34db83f6e (René Föhring   2020-05-10 16:41:39 +0200 123)       {:file_system, "~> 0.2.8"},

$ git log 34db83f6e
commit 34db83f6e068b09938fe231b2090d20124d51c37
Author: René Föhring <rene.foehring@5minds.de>
Date:   Sun May 10 16:41:39 2020 +0200

    WIP: Add basic file watcher
```

Here' s your answer: 34db83f6e068b09938fe231b2090d20124d51c37

**Comment by rrrene** [comment]
@skovmand @eksperimental There's also Credo `1.4.1` which should fix Elixir 1.11.0 compatibility for the current minor release. :+1:
